src/Sources.jl

```diff
@@ -98,6 +98,9 @@ end
 DirectorySource(path::String; target::String = "", follow_symlinks::Bool=false) =
     DirectorySource(path, target, follow_symlinks)
 
+# Try to guess if a URL is a Git repository
+isgitrepo(url::AbstractString) = endswith(url, ".git") || startswith(url, "git://")
```

Comment:
I defined this function so that we can use it in https://github.com/JuliaPackaging/BinaryBuilder.jl/blob/cf0121f82a09c8288f6cd3969b931d17e80b06aa/src/wizard/obtain_source.jl#L86, which should use the same logic